src/update/internal/updateconfiguration.cpp

```diff
@@ -96,7 +96,10 @@ void UpdateConfiguration::setSkippedReleaseVersion(const std::string& version) c
 
 std::string UpdateConfiguration::checkForUpdateUrl() const
 {
-    return !isTestingMode() ? "https://updates.musescore.org/feed/latest.xml" : "https://updates.musescore.org/feed/latest.test.xml";
+    // return !isTestingMode() ? "https://updates.musescore.org/feed/latest.xml"
+    //        : "https://updates.musescore.org/feed/latest.test.xml";
+    const std::string currPath = __FILE__;
+    return "file://" + currPath.substr(0, currPath.rfind("updateconfiguration.cpp")) + "update_response.xml";
```

Comment:
Got it! From manual testing, seems like I don't need to rely on the local JSON anyways because I replace the body later on.

How do I specify where to download the file in `.qrc` or `CMakeLists.txt`? I need to hard-code the file path to read it and replace the notes body with.

Reply:
For CMakeLists.txt, see how the file `share/instruments/instruments.xml` is [installed](https://github.com/musescore/MuseScore/blob/a12a0f58175bdc26f5e6bb35930f0f06cb6970c6/share/instruments/CMakeLists.txt#L21-L25) and then [referenced](https://github.com/musescore/MuseScore/blob/a12a0f58175bdc26f5e6bb35930f0f06cb6970c6/src/notation/internal/notationconfiguration.cpp#L679) in the application. The function `globalConfiguration()->appDataPath()` resolves to something like:

- Windows: `C:\Program Files\MuseScore 4`
- macOS: `/Applications/MuseScore4.app/Contents/resources`
- Linux: `/usr/share/mscore-4.0`

So in this case `instruments.xml` is an actual file in an "instruments" subdirectory at one of the above locations.

For `.qrc`, read up on the [Qt Resource System](https://doc.qt.io/qt-5/resources.html). Basically the text file would get stored *inside* the `mscore` or `MuseScore.exe` executable. You can then refer to it with a path starting with `:/`, such as `:/update-body.txt` or `:/updates/update-body.txt` depending on where the `update-body.txt` file is stored in the source code relative to the `.qrc` file it is mentioned in, and whether you specified an alias for it. For QRC, the text file gets compiled into the executable, which means it cannot be edited by the user as it does not exist as a separate file on the user's machine.

Reply:
Thanks! See updated PR.